### PR TITLE
[Improve][Transform] Add LLM model provider microsoft

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request

Add LLM model provider microsoft

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestLLMIT.java [hunk 1]
```diff
@@ -82,12 +84,25 @@ public void tearDown() throws Exception {
     }
 
     @TestTemplate
+    @DisabledOnContainer(
+            value = {},
+            type = {EngineType.SPARK, EngineType.FLINK})
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestLLMIT.java]: why?

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/ModelProvider.java [hunk 2]
```diff
@@ -26,6 +26,7 @@ public enum ModelProvider {
             "https://ark.cn-beijing.volces.com/api/v3/embeddings"),
     QIANFAN("", "https://aip.baidubce.com/rpc/2.0/ai_custom/v1/wenxinworkshop/embeddings"),
     KIMIAI("https://api.moonshot.cn/v1/chat/completions", ""),
+    MICROSOFT("", ""),
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/ModelProvider.java:29]: So when use microsoft model, user must configure api path? we should let users know this.

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestLLMIT.java (hunk 1 above)
Comment (corgy-w) [reply seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestLLMIT.java]: Forgot, I deleted it

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/ModelProvider.java (hunk 2 above)
Comment (corgy-w) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/ModelProvider.java:29]: get，I changes

File: docs/en/transform-v2/llm.md [hunk 3]
```diff
@@ -28,7 +28,9 @@ more.
 ### model_provider
 
 The model provider to use. The available options are:
-OPENAI, DOUBAO, KIMIAI, CUSTOM
+OPENAI, DOUBAO, KIMIAI, MICROSOFT, CUSTOM
+
+> tips: If you use Microsoft, please api_path cannot be empty
```
Comment (Hisoka-X) [inline docs/en/transform-v2/llm.md]:
```suggestion
> tips: If you use Microsoft, please make sure api_path cannot be empty
```

File: docs/zh/transform-v2/llm.md [hunk 4]
```diff
@@ -26,7 +26,9 @@
 ### model_provider
 
 要使用的模型提供者。可用选项为:
-OPENAI、DOUBAO、KIMIAI、CUSTOM
+OPENAI、DOUBAO、KIMIAI、MICROSOFT, CUSTOM
+
+> tips: 如果使用 Microsoft, api_path 配置不能为空
```
Comment (Hisoka-X) [inline docs/zh/transform-v2/llm.md]:
```suggestion
> tips: 如果使用 Microsoft, 请确保 api_path 配置不能为空
```

**Review by Hisoka-X — APPROVED** [review]

**Review by liugddx — APPROVED** [review]
+1

**Pull request merged**